### PR TITLE
collect metrics [T662]

### PR DESCRIPTION
**Pull request by Lagovas**
* collect metrics about connection count, time of processing connection, time of processing request/response to/from db, acrastruct decryption count
* small refactoring in acra-server. drop code duplication and move copy/paste logic to separate function

### CODE REVIEW

**Comment by vixentael** [comment]
(bitrise checks are still WIP, ignore them for now)

File: cmd/acra-server/listener.go [hunk 1]
```diff
@@ -168,97 +174,72 @@ func (server *SServer) handleConnection(connection net.Conn) {
...
-		} else {
-			log.Infof("Got new connection to AcraServer: %v", connection.RemoteAddr())
-		}
-		go func() {
-			server.cmACRA.Incr()
-			server.handleConnection(connection)
-			server.cmACRA.Done()
-		}()
-	}
+	server.addListener(listenerWithFileDescriptor)
```
Comment (vixentael) [inline cmd/acra-server/listener.go]: 👍

File: cmd/acra-server/prometheus.go [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+const (
+	connectionTypeLabel = "connection_type"
+	apiConnectionType   = "api"
+	dbConnectionType    = "db"
+)
+
+var (
+	connectionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "acraserver_connection_total",
```
Comment (vixentael) [inline cmd/acra-server/prometheus.go]: `connections`?

File: cmd/acra-connector/prometheus.go [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+const (
+	connectionTypeLabel = "connection_type"
+	apiConnectionType   = "api"
+	dbConnectionType    = "db"
+)
+
+var (
+	connectionCounter = prometheus.NewCounterVec(
+		prometheus.CounterOpts{
+			Name: "acraconnector_connection_total",
```
Comment (vixentael) [inline cmd/acra-connector/prometheus.go]: `connections`?

**Comment by vixentael** [comment]
1.
>  time of processing request/response to/from db

we collect only time of success requests/response? if f.e. TLS handshake failed, we don't collect this time?

2. 
> acrastruct decryption

currently we don't collect poison record checks? in future I would add type `DecryptionTypePoisonRecordFound`

**Comment by Lagovas** [comment]
> we collect only time of success requests/response? if f.e. TLS handshake failed, we don't collect this time?

yes, now we track only success results

> currently we don't collect poison record checks? in future I would add type 

will add in a future

> acraconnector_connections_total

updated

**Pull request merged**